plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,349 @@ var plugin = {},
+        if (params.qstring.popup_comment_callout) changes["popup_comment_callout"] = params.qstring.popup_comment_callout;
+        if (params.qstring.popup_thanks_message) changes["popup_thanks_message"] = params.qstring.popup_thanks_message;
+        if (params.qstring.trigger_position) changes["trigger_position"] = params.qstring.trigger_position;
+        if (params.qstring.trigger_bg_color) changes["trigger_bg_color"] = params.qstring.trigger_bg_color;
+        if (params.qstring.trigger_button_text) changes["trigger_button_text"] = params.qstring.trigger_button_text;
+        if (params.qstring.trigger_font_color) changes["trigger_font_color"] = params.qstring.trigger_font_color;
+        if (params.qstring.target_devices) changes["target_devices"] = params.qstring.target_devices;
+        if (params.qstring.target_page) changes["target_page"] = params.qstring.target_page;
+        if (params.qstring.target_pages) changes["target_pages"] = params.qstring.target_pages || [];
+        if (params.qstring.is_active) changes["is_active"] = params.qstring.is_active;
+        if (params.qstring.hide_sticker) changes["hide_sticker"] = params.qstring.hide_sticker;
+
+        common.db.collection(collectionName)
+            .findAndModify({_id:common.db.ObjectID(id)},{}, changes,
+                function(err, widget) {
+                    if (!err) {
+                        common.returnMessage(ob.params, 200, 'Widget updated');
+                        return true;
+                    } else {
+                        common.returnMessage(ob.params, 500, err.message);
+                        return false;
+                    }
+                })
+        return true;
+    }
+
+    var removeWidgetData = function(widgetId, app, callback) {
+        var collectionName = "feedback" + app;
+
+        common.db.collection(collectionName)
+            .remove({"widget_id":widgetId},
+            function(err) {
+                if (!err) callback(null);
+                else callback(err);
+            })
+    }
+
+    plugins.register("/i", function(ob){
+        var params = ob.params;
+        if (params.qstring.events && params.qstring.events.length && Array.isArray(params.qstring.events)) {
+            params.qstring.events = params.qstring.events.filter(function(currEvent){
+                if (currEvent.key == "[CLY]_star_rating"){
+                    
+                    var collectionName = 'feedback' + params.qstring.app_key;
+
+                    common.db.collection(collectionName)
+                        .insert({
+                            "email": currEvent.segmentation.email,
+                            "comment": currEvent.segmentation.comment,
+                            "ts":currEvent.timestamp,
+                            "device_id":params.qstring.device_id,
+                            "cd":new Date(),
+                            "uid":params.app_user.uid,
```

Comment:
This collection should have 2 indexes on uid field and on ts field. Indexes should be added on plugin enabling and on app lifecycle. Comment on app life cycle is at the end of the file